### PR TITLE
Slices proposal

### PR DESCRIPTION
**Pull request by joshring**
My first PR, so apologies if I am not following conventions please correct me. 

### Proposal for slices
- Fixed the examples borrowing from the "differences from C" page.
- Focused more runnable example functions.
- Made the content a bit more long form to make it clearer for more unfamiliar syntax to reduce confusion. 
- Splitting some parts into two to make it easier to digest where appropriate.
- Added a bit more vertical spacing where the ideas being shown were noticeably different

### CODE REVIEW

File: src/content/docs/references/docs/arrays.md [hunk 1]
```diff
@@ -14,84 +14,132 @@ These are declared as `<type>[<size>]`, e.g. `int[4]`. Fixed arrays are treated
...
+The final type is the slice `<type>[]`  e.g. `int[]`. A slice is a view into either a fixed or variable array. Internally it is represented as a struct containing a pointer and a size. Both fixed and variable arrays may be converted into slices, and slices may be implicitly converted to pointers.
+
+```c3
+fn void test() 
+{
     int[3] a = { 1, 2, 3 };
-    int[*] b = { 4, 5, 6 }; // Type inferred to be int[3]
 
-## Slice
+    int[4]* b = (int[4]*)&a;    // Explicit cast required
```
Comment (lerno) [inline src/content/docs/references/docs/arrays.md]: This example should be changed, it's under "slice" but shows no slices at all... The old one was better.

File: src/content/docs/references/docs/arrays.md [hunk 2]
```diff
@@ -14,84 +14,132 @@ These are declared as `<type>[<size>]`, e.g. `int[4]`. Fixed arrays are treated
...
     int[] c = a[..4];
     int[] d = a[0..];
     int[] e = a[..];
+    
     int[] f = a[0:5];
     int[] g = a[:5];
+}
+```
+
+You can also slice referencing the last index, where
```
Comment (lerno) [inline src/content/docs/references/docs/arrays.md]: A better way would be to describe ^i as meaning `len - i`. It's both applicable to `:` and `..` syntax, but used differently. For example getting the entire slice with `^` would be `a[0..^1]` or `a[0:^0]`. This needs to be explained I think.

**Review by lerno — CHANGES_REQUESTED** [review]

File: src/content/docs/references/docs/arrays.md (hunk 2 above)
Comment (joshring) [reply src/content/docs/references/docs/arrays.md]: Thanks I now realised my mistake with `^i` is  `len-i` that helped a lot, in latest commit

File: src/content/docs/references/docs/arrays.md (hunk 1 above)
Comment (joshring) [reply src/content/docs/references/docs/arrays.md]: I went through the conversion table to make some nice examples and documented it, hopefully that's better now, it helped my understanding too :)

**Comment by lerno** [comment]
Let me accept this and then I'll do some additional tweaks afterwards. Thank you for the help!

**Pull request merged**